specifyweb/specify/migrations/0002_geo.py

```diff
@@ -149,18 +150,19 @@ def revert_default_collection_object_types():
             cog_type_picklist.delete()
 
 def set_discipline_for_taxon_treedefs():
-    for treedef in Taxontreedef.objects.all():
-        if treedef.discipline:
-            continue
-        cot = Collectionobjecttype.objects.filter(taxontreedef=treedef).first()
-        if cot:
-            treedef.discipline = cot.collection.discipline
-            treedef.save()
+    taxon_treedefs_without_discipline = Taxontreedef.objects.filter(discipline__isnull=True)
+    collection_object_types = Collectionobjecttype.objects.filter(taxontreedef__in=taxon_treedefs_without_discipline)
```

Comment:
(optional)
Can you combine these into a single query? 
Something like 

```suggestion
    collection_object_types = Collectionobjecttype.objects.filter(taxontreedef__discipline__isnull=True)
```